### ISSUE TITLE
[8.0] cannot cancel PO containing only service product and with invoicing control = 'Based on incoming shipments'

### ISSUE THREAD

**Issue by yostashiro** [odoo/odoo#3492]
Impacted versions:
* 8.0

Steps to reproduce:
1. create a purchase order only with service product(s) and Invoicing Control = 'Based on incoming shipments'
2. press 'Cancel' button

Current behavior:
* nothing happens (the PO status remains as 'Purchase Confirmed')

Expected behavior:
* PO status is changed to 'Cancelled'

User should not be processing PO with Invoicing Control 'Based on incoming shipments' if it only contains service products.  However, the problem is that there is no way to fix the PO if you forget to correct the Invoicing Control.

**Comment by jbeficent** [comment]
This is also happening in version 7.0

**Comment by jbeficent** [comment]
This Pull Request introduces a module that prevents this from happening, and allows you to cancel wrong PO's. https://github.com/OCA/purchase-workflow/pull/89

**Comment by jbeficent** [comment]
@jco-odoo The module adds a check that prevents a user from setting the invoice control 'Based on incoming shipments' if the PO contains an item that is not stockable.

@odony I will then prepare a PR to submit to core. The module also introduces the possibility to cancel PO's that were left inconsistent due to this issue. Should I include that in the PR? Supposedly, once the new validation is in place there will be no need for such function.

**Comment by odony** [comment]
@jbeficent Thanks! I'd leave out the workflow change as it should really be unnecessary once old cases are fixed. A single SQL query could be used to fix at once all purchase orders that are stuck, I can provide the query if it helps.

**Comment by jbeficent** [comment]
Thanks @odony. I closed the previous PR. Will submit new one. The query would help ;)

**Comment by jbeficent** [comment]
@odony Now submitted PR.

**Comment by dsabrinarg** [comment]
We are being affected by this issue!

**Comment by jco-odoo** [comment]
Would you be so kind to pass me the number of this new PR?

**Comment by jbeficent** [comment]
@jco-odoo PR #5370

**Comment by dmo-odoo** [comment]
Hello,

I cannot reproduce this bug on runbot, either in 7.0 or in 8.0 by following the above steps. I noticed that the purchase order was correctly cancelled, even though it only contained a service product. Could you please give it a try ?

Best regards,
David

**Comment by dmo-odoo** [comment]
After a long discussion with jco, we decided to adopt the same behavior in 7.0 and 8.0:
https://github.com/odoo/odoo/commit/6758b4cfc5ab5a9bf54c30aa14ed2dfd2fbf534d
